File: app/src/main/java/im/dacer/kata/core/action/JishoSearchAction.java
/*
 * The MIT License (MIT)
 * Copyright (c) 2016 baoyongzhang <baoyz94@gmail.com>
 */
package im.dacer.kata.core.action;

import android.net.Uri;

/**
 * Created by baoyongzhang on 2016/10/26.
 */
public class JishoSearchAction extends SearchAction {

    public static JishoSearchAction create() {
        return new JishoSearchAction();
    }

    @Override
    public Uri createSearchUriWithEncodedText(String encodedText) {
        return Uri.parse("http://jisho.org/search/" + encodedText);
    }
}

File: app/src/main/java/im/dacer/kata/util/action/JishoSearchAction.kt
package im.dacer.kata.util.action

import android.net.Uri

class JishoSearchAction : WebSearchAction() {

    override fun createSearchUriWithEncodedText(encodedText: String?): Uri {
        return Uri.parse("http://jisho.org/start/" + encodedText!!)
    }

    companion object {

        fun create(): JishoSearchAction {
            return JishoSearchAction()
        }
    }
}
